android/app/src/main/java/org/fossasia/openevent/activities/TrackSessionsActivity.java

```diff
@@ -134,6 +143,33 @@ private void loadData() {
 
             mSessions.clear();
             mSessions.addAll(track.getSessions().sort("startsAt"));
+
+//            finding upcoming and ongoing sessions
+            Date current;
+            int countUpcoming = 0;
+            int countOngoing = 0;
+            for (Session trackSession : mSessions) {
+                flag = 0;
+                Date start = new Date();
+                Date end = new Date();
+                try {
+                    start = DateUtils.getDate(trackSession.getStartsAt());
+                    end = DateUtils.getDate((trackSession.getEndsAt()));
+                } catch (ParseException e) {
+                    e.printStackTrace();
+                }
+                current = new Date();
+                if (((start.before(current)) || (start.equals(current))) && ((end.after(current)) || (end.equals(current)))) {
+                    ongoingPosition = countUpcoming;
+                    break;
+                } else if (start.after(current)) {
+                    upcomingPosition = countOngoing;
+                    break;
+
+                } else flag += 1;
+                countUpcoming += 1;
+                countOngoing += 1;
```

Comment:
Which initialisations are you talking about? `flag`, `countUpcoming` and `countOngoing`? I think they are still required. The IDE gives an initialisation error otherwise.